test/services/LookupServiceSpec.scala

```diff
@@ -16,47 +16,74 @@
+import play.api.Configuration
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class LookupServiceSpec extends ServiceBaseSpec {
+class LookupServiceSpec extends ServiceBaseSpec with MockAppConfig {
+
+  val nino: String                             = "AA123456A"
+  val mtdId                                    = "some id"
+  val ifsReference: MtdIdIfsReference          = MtdIdIfsReference(mtdId)
+  val desReference: MtdIdDesReference          = MtdIdDesReference(mtdId)
+  val reference: MtdIdResponse                 = MtdIdResponse(mtdId)
+  val cached: MtdIdCached                      = MtdIdCached(nino, mtdId)
+  val lookupCacheResponse: Option[MtdIdCached] = None
+  val isCachedResponse: Boolean                = true
 
   trait Test extends MockBusinessDetailsConnector with MockLookupRepository {
-    lazy val target = new LookupService(mockBusinessDetailsConnector, mockLookupRepository)
+    lazy val target = new LookupService(mockBusinessDetailsConnector, mockLookupRepository, mockAppConfig)
+
   }
 
-  "calling .getMtdId function" when {
+  "calling .getMtdIdFromDes function" when {
+    "a known MTD Nino is passed" should {
+      "return a valid mtdId" in new Test {
+        MockedAppConfig.featureSwitches
+          .returns(Configuration("ifs.enabled" -> false))
+          .anyNumberOfTimes()
+
+        mockGetMtdIdFromDes(nino).returns(Future.successful(Right(ResponseWrapper(correlationId, desReference))))
+        MockedLookupRepository.save(cached).returns(Future.successful(isCachedResponse))
+        MockedLookupRepository.getMtdReference(nino).returns(Future.successful(lookupCacheResponse))
+
+        private val result = await(target.getMtdId(nino))
+
+        result shouldBe Right(reference)
+      }
+    }
+  }
+
+  "calling .getMtdIdFromIfs function" when {
 
-    val nino: String = "AA123456A"
-    val mtdId        = "some id"
     "a known MTD NINO is passed and not available in lookup cache" should {
       "save a valid mtdId in the lookup cache and return" in new Test {
 
-        val connectorResponse   = Right(mtdId)
-        val lookupCacheResponse = None
-        val isCachedResponse    = true
+        MockedAppConfig.featureSwitches
+          .returns(Configuration("ifs.enabled" -> true))
+          .anyNumberOfTimes()
 
-        MockedLookupRepository.save(nino, mtdId).returns(Future.successful(isCachedResponse))
-        mockGetMtdId(nino).returns(Future.successful(connectorResponse))
         MockedLookupRepository.getMtdReference(nino).returns(Future.successful(lookupCacheResponse))
+        mockGetMtdIdFromIfs(nino).returns(Future.successful(Right(ResponseWrapper(correlationId, ifsReference))))
+        MockedLookupRepository.save(cached).returns(Future.successful(isCachedResponse))
 
         private val result = await(target.getMtdId(nino))
 
-        result shouldBe connectorResponse
+        result shouldBe Right(reference)
```

Comment:
Sorry when I was suggesting in-lining the `val`s I meant pulling the values of them into the tests so that they could be read in isolation. 

E.g. something like this makes it clear what the cache is returning (None) and that the same mtdIds are used throughout:
```scala  
        MockedAppConfig.featureSwitches
          .returns(Configuration("ifs.enabled" -> true))
          .anyNumberOfTimes()

        MockedLookupRepository.getMtdReference(nino).returns(Future.successful(None))
        mockGetMtdIdFromIfs(nino).returns(Future.successful(Right(ResponseWrapper(correlationId, MtdIdIfsReference(mtdId)))))
        MockedLookupRepository.save(MtdIdCached(nino, mtdId)).returns(Future.successful(true))

        await(target.getMtdId(nino)) shouldBe Right(MtdIdResponse(mtdId))
```